workflows/snvphylnfc.nf

```diff
@@ -67,34 +98,118 @@ workflow SNVPHYL {
                fastq_2 ? tuple(meta, [ file(fastq_1), file(fastq_2) ]) :
                tuple(meta, [ file(fastq_1) ])}
 
-    ASSEMBLY_STUB (
-        input
+    INDEXING(
+        params.refgenome
+    )
+    ch_versions = ch_versions.mix(INDEXING.out.versions)
+
+    FIND_REPEATS(
+        params.refgenome
+    )
+    ch_versions = ch_versions.mix(FIND_REPEATS.out.versions)
+
+    SMALT_MAP(
+        input, INDEXING.out.ref_fai, INDEXING.out.ref_sma, INDEXING.out.ref_smi
     )
-    ch_versions = ch_versions.mix(ASSEMBLY_STUB.out.versions)
+    ch_versions = ch_versions.mix(SMALT_MAP.out.versions)
 
-    // A channel of tuples of ({meta}, [read[0], read[1]], assembly)
-    ch_tuple_read_assembly = input.join(ASSEMBLY_STUB.out.assembly)
+    SORT_INDEX_BAMS(
+        SMALT_MAP.out.bams
+    )
+    ch_versions = ch_versions.mix(SORT_INDEX_BAMS.out.versions)
+
+    // TODO: Review this module further (purpose, version, code injection, container, etc.)
+    GENERATE_LINE_1(
```

Comment:
It seems that `GENERATE_LINE_1` and `GENERATE_LINE_2` are used to generate command-line strings for use later in running commands.

I'm pretty sure this could be done directly in Groovy as well. Could you look into this? This would cut out having to run these processes.

Reply:
Yeah, those two modules are the ones I'm most nervous about, but only partially understand. They also seem like a potential weak spot for code injection. I'll have to dig into them deeper.

Reply:
Sounds good. Thanks for looking into this.